### DISCUSSION TITLE
apache-tomcat: Providing Additional JARs to Tomcat

### DISCUSSION THREAD

**Discussion by ZBSTooling** [paketo-buildpacks/feedback#221] [Q&A] [answered]
I have to add a jar to tomcat's lib directory. Is there an example how to use the described feature?
[Providing Additional JARs to Tomcat](https://github.com/paketo-buildpacks/apache-tomcat#providing-additional-jars-to-tomcat)

**Comment by dmikusa** [comment, marked as answer]
> Buildpacks can contribute JARs to the CLASSPATH of Tomcat

This is for other Buildpacks to add JARs. I think you could technically use it without a buildpack, but it is not the intended way to do this.

As a user, you would want [External Configuration](https://github.com/paketo-buildpacks/apache-tomcat#external-configuration-package). This allows you to overlay basically anything on top of the Tomcat install. Think of it as your external configuration bundle is just extracted on top of Tomcat. The example there shows config, but you could just as easily overlay a JAR file into the `lib/` directory.

**Reply by ZBSTooling** [reply]
External Configuration is working for jars in tomcat's lib folder. Maybe it should be documented 🤔
